tokendito/helpers.py

```diff
@@ -416,40 +435,50 @@ def process_okta_aws_app_url(app_url=None):
     :param app_url: string with okta tile URL.
     :return: None.
     """
-    if app_url is None:
-        logging.error(
-            "Okta Application URL not found in profile '{}'.\nPlease verify your options"
-            " or re-run this application with the --configure flag".format(settings.okta_profile))
-        sys.exit(2)
-    # Prepare final Okta and AWS app Url
-    url = urlparse(app_url)
-
-    if url.scheme == '' or url.netloc == '' or url.path == '':
-        logging.error("Okta Application URL invalid. Please check your configuration"
-                      " and try again.")
+    if not validate_okta_aws_app_url(settings.okta_aws_app_url):
+        logging.error("Okta Application URL not found, or invalid. Please check "
+                      "your configuration and try again.")
         sys.exit(2)
 
+    url = urlparse(settings.okta_aws_app_url)
     okta_org = '{}://{}'.format(url.scheme, url.netloc)
     okta_aws_app_url = '{}{}'.format(okta_org, url.path)
     setattr(settings, 'okta_org', okta_org)
     setattr(settings, 'okta_aws_app_url', okta_aws_app_url)
 
 
-def user_configuration_input():
+def user_configuration_input(url='', username=''):
```

Comment:
since we arent passing any params, use locals instead and remove those params? (if we keep the params and in the future pass them it will change the functionality (since it will then not enter the while loop.

Reply:
fixed.